homework-submission/React_HW2/todoApp2.js

```diff
@@ -0,0 +1,149 @@
+'use strict';
+
+// ContentDynamicTodo component is for map function to use
+const ContentDynamicTodo = ({ description, deadline, className, deleteItem }) => {
+  return (
+    <tbody>
+      <tr className={className} onClick={deleteItem}>
+        <td class="evenBackgroundColor">
+          <td class="content">{description}</td>
+          <td class="content">{deadline}</td>
+          <td class="content">{className}</td>
+        </td>
+      </tr>
+    </tbody>
+  );
+};
+
+// DynamicList where map function is
+const DynamicList = ({ list, deleteItem }) => {
+  const runTodoList = list.map(todo => (
+    <ContentDynamicTodo
+      key={todo.id}
+      description={todo.description}
+      deadline={todo.deadline}
+      className={todo.done ? '√' : 'x'}
+      deleteItem={() => {
+        deleteItem(todo.id);
+      }}
+    />
+  ));
+  return <tbody>{runTodoList}</tbody>;
+};
+
+// Here is our main Application component as a class
+class AppTodo2 extends React.Component {
+  //TodoList which contains new item as state
+  constructor(props) {
+    super(props);
+    this.state = {
+      todoList: [
+        {
+          id: 1,
+          description: 'Get out of bed',
+          deadline: '2017-09-11',
+          done: true,
+        },
+        {
+          id: 2,
+          description: 'Brush teeth',
+          deadline: '2017-09-10',
+          done: false,
+        },
+        {
+          id: 3,
+          description: 'Eat breakfast',
+          deadline: '2017-09-09',
+          done: false,
+        },
+      ],
+      newTodo: {
+        id: '',
+        description: 'Description',
+        deadline: 'No Deadline Entry',
+        done: false,
+      },
+    };
+  }
+
+  // Adding a description for new todo.
+  NewTodoDescription = () => {
+    return (
+      <p>
+        <label htmlFor="">Description </label>
```

Comment:
If no need for it, you can remove it!

```suggestion
        <label>Description </label>
```

Reply:
I used it for CSS.

Reply:
**Strange I don't see it there!** 😕 

How ever, you could use classes or an ids instead, using htmlFor only for the purpose of css doesn't make it a good use.

Reply:
Ok I implement it